Tools/glib/apply-build-revision-to-files.py

```diff
@@ -76,13 +76,20 @@ def main(args):
                     if line.startswith("revision"):
                         line = "revision=@BUILD_REVISION@\n"
                     lines.append(line)
-            data = "".join(lines)
+            template = "".join(lines)
         else:
             print(f"Support for expanding @BUILD_REVISION@ in {file!s} is missing.")
             return 1
 
-        with file.open('w') as fd:
-            fd.write(data.replace('@BUILD_REVISION@', build_revision))
+        new_contents = template.replace('@BUILD_REVISION@', build_revision)
+        # Only write the new contents to the output file if there are any changes.
+        # Otherwise, the file would become 'dirty' and all dependents would be
+        # recompiled unnecessarily.
+        with file.open() as fd:
+            old_contents = fd.read()
+        if old_contents != new_contents:
+            with file.open('w') as fd:
+                fd.write(new_contents)
```

Comment:
Nice, good catch!

Reply:
I just realized this will fail if the header file doesn't exist when run.

Reply:
nvm, this will not occur because the file is first created as just a copy of the template by cmake.

```cmake
configure_file(Shared/glib/BuildRevision.h.in ${FORWARDING_HEADERS_WPE_DIR}/BuildRevision.h)
```